src/modules/acbs/acbs-party.service.get-party-by-search-text.test.ts

```diff
@@ -0,0 +1,144 @@
+import { HttpService } from '@nestjs/axios';
+import { GetPartyGenerator } from '@ukef-test/support/generator/get-party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { AcbsPartyService } from './acbs-party.service';
+import { AcbsException } from './exception/acbs.exception';
+
+describe('AcbsPartyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+
+  let httpService: HttpService;
+  let service: AcbsPartyService;
+
+  let httpServiceGet: jest.Mock;
+
+  beforeEach(() => {
+    httpService = new HttpService();
+
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    service = new AcbsPartyService({ baseUrl }, httpService);
+  });
+
+  describe('getPartyBySearchText', () => {
+    const searchText = 'searchText';
```

Comment:
Can we add assertions of searchText with other void values please?

* `123123`
* !"£!"£
* `  `

Reply:
In the ACBS service we mock what is returned given this text so the only place we can test these are in the API tests.
We cover `   `  (bad request) and `123123` (valid) already, however we don't have tests for non-alphanumeric characters as it's a valid search in ACBS so we have not added validation for this